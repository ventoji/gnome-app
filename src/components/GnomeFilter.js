import React from 'react';
import SvgIcon from './SvgIcon';

const GnomeFilter = ({text, setTextFilter}) => {

    const onTextChange = (e) => {
        setTextFilter(e.target.value);
    }; 

    return (
  
        <div className="gnome-form__input">
          <input
            type="text"
            className="gnome-form__input-text"
            placeholder="Try finding some gnome"
            disabled={ false}
            value={text}
            onFocus={(e) => e.target.placeholder = ''} 
            onBlur={(e) => e.target.placeholder = 'Try finding some gnome'} 
            onChange={onTextChange}
          />
          <button className="gnome-form__button">
          <SvgIcon svgClass="gnome-form__icon" svgName="img/sprite.svg#icon-magnifying-glass" />
          </button>

        </div>
    );
};

export default GnomeFilter;