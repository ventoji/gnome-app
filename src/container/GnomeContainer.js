import {connect} from 'react-redux';
import Gnome from '../components/Gnome';
import { withRouter } from 'react-router-dom';

export const mapStateToProps = state => ({
        ...state.gnome
});

const GnomeContainer = connect(mapStateToProps)(Gnome);

export default withRouter(GnomeContainer);

