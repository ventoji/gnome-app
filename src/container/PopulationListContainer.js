import {connect} from 'react-redux';
import { addGnomeItem } from '../actions/gnome';
import { addGnomes } from '../actions/gnomes';
import { bindActionCreators } from 'redux';
import PopulationList from '../components/PopulationList';
import selectGnomes from '../selectors/gnome';


const mapStateToProps = (state, props) => {

    return {
      gnome: state.gnome,
      filters: state.filters,
      gnomes: selectGnomes(state.gnomes, state.filters)
    };
  };

const mapDispatchToProps = (dispatch, props) =>
    bindActionCreators({ addGnomeItem, addGnomes} , dispatch);

const PopulationListContainer = connect(mapStateToProps,mapDispatchToProps)(PopulationList);

export default PopulationListContainer;