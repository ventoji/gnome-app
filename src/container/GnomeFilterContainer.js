import {connect} from 'react-redux';
import { setTextFilter } from '../actions/filter';
import { bindActionCreators } from 'redux';
import GnomeFilter from '../components/GnomeFilter';


export const mapStateToProps = state => ({
  state: state.filters
});

export const mapDispatchToProps = dispatch =>
    bindActionCreators({ setTextFilter} , dispatch);

const GnomeFilterContainer = connect(mapStateToProps,mapDispatchToProps)(GnomeFilter);

export default GnomeFilterContainer;