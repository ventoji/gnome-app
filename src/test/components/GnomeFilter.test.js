import { mapStateToProps } from '../../container/GnomeFilterContainer';
import  GnomeFilter  from '../../components/GnomeFilter';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

let wrapper, tree, setTextFilter;

describe(`The Gnome Filter Component`, () => {

    describe(`The Container Element`, () => {

        describe(`mapStateToProps`, () => {
            it("should map the state to props correctly", () => {
       
                const state = {
                    filters: 'Tobus'
                };
                const componentState = mapStateToProps(state);
                expect(componentState).toEqual({
                    state: 'Tobus'});

            });

        });

    });

    describe(`The display element`, () => {

        beforeEach(() => {
            setTextFilter = jest.fn();
            wrapper = shallow( <GnomeFilter  />);
             tree = renderer
            .create(<GnomeFilter  />)
            .toJSON(); 
        });
       

        it('Should render component properly', () => {
            expect(tree).toMatchSnapshot();
        });

        it('Should render component properly with all data correctly', () => {
             wrapper.setProps({
                filters: 'Tobus'
            });
            expect(wrapper).toMatchSnapshot(); 
        });

        it('Should handle text change on input  ', () => {
             const value='Tobus';
            wrapper.setProps({
                setTextFilter:  setTextFilter
            });
            wrapper.find('input').simulate('change', {
                target: {value}
            });
            expect(setTextFilter).toHaveBeenLastCalledWith(value); 
        
        });
    });


});