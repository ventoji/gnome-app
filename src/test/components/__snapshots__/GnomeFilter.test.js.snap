// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Gnome Filter Component The display element Should render component properly 1`] = `
<div
  className="gnome-form__input"
>
  <input
    className="gnome-form__input-text"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Try finding some gnome"
    type="text"
  />
  <button
    className="gnome-form__button"
  >
    <svg
      className="gnome-form__icon"
    >
      <use
        xlinkHref="/img/sprite.svg#icon-magnifying-glass"
      />
    </svg>
  </button>
</div>
`;

exports[`The Gnome Filter Component The display element Should render component properly with all data correctly 1`] = `
<div
  className="gnome-form__input"
>
  <input
    className="gnome-form__input-text"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Try finding some gnome"
    type="text"
  />
  <button
    className="gnome-form__button"
  >
    <SvgIcon
      svgClass="gnome-form__icon"
      svgName="img/sprite.svg#icon-magnifying-glass"
    />
  </button>
</div>
`;
